examples/client-server/main.go

```diff
@@ -0,0 +1,200 @@
+	optionClientThreads := flag.Int("threads", 3, "How many parallel clients should be there trying to reach the servers behind the LB")
+
+	// Flag for launching test servers
+	optionTestServers := flag.Bool("start-test-servers", false, "Launch test servers to be hidden behind the load balancer, must be in conjunction with --routes xxx:port,xxx:port")
+
+	flag.Parse()
+
+	// Prepare routes from flags
+	routes := strings.Split(*routesStr, ",")
+	pRoutes := make([]xlb.Route, 0)
+	for _, rte := range routes {
+		if len(rte) == 0 {
+			continue
+		}
+		pRoutes = append(pRoutes, testUtil.TestServicePoolRoute{
+			ServicePath:   rte,
+			ServiceActive: true,
+		})
+	}
+
+	// Generate TLS data in folder
+	if *optionGenTLS {
+		err := testUtil.CreateLocalTLSData(*identity)
+		if err != nil {
+			log.Fatalf("cannot generate TLS data, error: %+v", err)
+		}
+	}
+
+	// Clean TLS data in folder
+	if *optionWipeTLS {
+		removed, err := testUtil.WipeLocalTLSData("./")
+		if err != nil {
+			log.Fatalf("cannot generate TLS data, error: %+v", err)
+		}
+		log.Printf("files deleted: %+v", removed)
+	}
+
+	// Launch load balancer using provided options
+	if *optionLoadBalancer {
+
+		if len(pRoutes) == 0 {
+			log.Fatalf("no routes provided")
+			return
+		}
+
+		cert, err := os.ReadFile(*serverCert)
+		if err != nil {
+			log.Fatalf("server certificate not found")
+			return
+		}
+
+		key, err := os.ReadFile(*serverKey)
+		if err != nil {
+			log.Fatalf("server key not found")
+			return
+		}
+
+		balancer, err := xlb.NewLoadBalancer(ctx, []xlb.ServicePool{testUtil.TestServicePoolConfig{
+			SvcIdentity:          "test",
+			SvcPort:              *port,
+			SvcRateQuotaTimes:    *rateQuota,
+			SvcRateQuotaDuration: *rateDuration,
+			SvcRoutes:            pRoutes,
+			Certificate:          string(cert),
+			CertKey:              string(key),
+		}}, xlb.Options{LogLevel: "debug"})
+		if err != nil {
+			return
+		}
+
+		// Listen for all threads
+		err = balancer.Listen()
+		if err != nil {
+			log.Fatalf("listen returned error: %+v", err)
+			return
+		}
+	}
+
+	// Launch the client to send the message
+	if *optionClient {
+		launchClient(*port, *optionClientThreads)
+	}
+
+	// Launch the test servers
+	if *optionTestServers {
+
+		if len(pRoutes) == 0 {
+			log.Fatalf("no routes provided")
+			return
+		}
+
+		err := launchTestServers(ctx, pRoutes)
+		if err != nil {
+			fmt.Printf("Error launching test servers: %v\n", err)
+		}
+	}
+}
+
+func launchClient(targetPort int, threads int) {
+	// Test load balancer
+	wg := sync.WaitGroup{}
+	for i := 0; i < threads; i++ {
+		wg.Add(1)
+		go func() {
+			// Test load balancer
+			res, err := testUtil.SendRequest("https://localhost:" + strconv.Itoa(targetPort) + "/api")
+			if err != nil {
+				log.Printf(fmt.Errorf("cannot reach remotes, error: %w", err).Error())
+			} else {
+				log.Println(res)
+			}
+			wg.Done()
```

Comment:
Suggestion: use a defer to ensure that Done is always called. While fine now if either branch adds a return in the future the waitgroup will block forever.
```suggestion
			defer wg.Done()
			// Test load balancer
			res, err := testUtil.SendRequest("https://localhost:" + strconv.Itoa(targetPort) + "/api")
			if err != nil {
				log.Printf(fmt.Errorf("cannot reach remotes, error: %w", err).Error())
			} else {
				log.Println(res)
			}
```